components/conversations/EmojiReactionPicker.js

```diff
@@ -0,0 +1,265 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useMutation } from '@apollo/client';
+import { Manager, Popper, Reference } from 'react-popper';
+import styled, { css } from 'styled-components';
+
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+import useGlobalBlur from '../../lib/hooks/useGlobalBlur';
+
+import { Flex } from '../Grid';
+import AddReactionIcon from '../icons/AddReactionIcon';
+import StyledButton from '../StyledButton';
+import StyledCard from '../StyledCard';
+import StyledRoundButton from '../StyledRoundButton';
+
+const addReactionMutation = gqlV2/* GraphQL */ `
+  mutation AddEmojiReaction($emoji: String!, $update: UpdateReferenceInput, $comment: CommentReferenceInput) {
+    addEmojiReaction(emoji: $emoji, update: $update, comment: $comment) {
+      update {
+        id
+        reactions
+        userReactions
+      }
+      comment {
+        id
+        reactions
+        userReactions
+      }
+    }
+  }
+`;
+
+const removeReactionMutation = gqlV2/* GraphQL */ `
+  mutation RemoveEmojiReaction($emoji: String!, $update: UpdateReferenceInput, $comment: CommentReferenceInput) {
+    removeEmojiReaction(emoji: $emoji, update: $update, comment: $comment) {
+      update {
+        id
+        reactions
+        userReactions
+      }
+      comment {
+        id
+        reactions
+        userReactions
+      }
+    }
+  }
+`;
+
+const Emoji = styled.div`
+  font-size: 15px;
+`;
+
+const ReactionButton = styled(StyledRoundButton).attrs({ isBorderless: true, buttonSize: 'small' })`
+  margin: 4px;
+  background: white !important;
+
+  ${Emoji} {
+    transition: transform 0.15s cubic-bezier(0.2, 0, 0.13, 2);
+  }
+
+  &:hover {
+    ${Emoji} {
+      transform: scale(1.3);
+    }
+  }
+
+  ${props =>
+    props.isSelected &&
+    css`
+      background: ${props.theme.colors.primary[200]} !important;
+    `}
+`;
+
+const getOptimisticResponse = (entity, emoji, isAdding, isComment) => {
+  const userReactions = entity.userReactions || [];
+  if (isAdding) {
+    const newCount = (entity.reactions[emoji] || 0) + 1;
+
+    let entityObj;
+    if (isComment) {
+      entityObj = {
+        comment: {
+          __typename: entity.__typename,
+          id: entity.id,
+          reactions: { ...entity.reactions, [emoji]: newCount },
+          userReactions: [...userReactions, emoji],
+        },
+      };
+    } else {
+      entityObj = {
+        update: {
+          __typename: entity.__typename,
+          id: entity.id,
+          reactions: { ...entity.reactions, [emoji]: newCount },
+          userReactions: [...userReactions, emoji],
+        },
+      };
+    }
+
+    return {
+      __typename: 'Mutation',
+      addEmojiReaction: {
+        __typename: 'addEmojiReaction',
+        ...entityObj,
+      },
+    };
```

Comment:
This can be simplified with the approach previously mentioned

```suggestion
    const { __typename } = entity;
    const fieldName = __typename === 'Update' ? 'update' : 'comment';
    const otherFieldName = fieldName === 'comment' ? 'update' : 'comment';

    return {
      __typename: 'Mutation',
      addEmojiReaction: {
        __typename: 'addEmojiReaction',
        [otherFieldName]: null,
        [fieldName]: {
          __typename,
          id: entity.id,
          reactions: { ...entity.reactions, [emoji]: newCount },
          userReactions: [...userReactions, emoji],
        },
      },
    };
```

Reply:
Done. Thanks. This looks much nicer. 😄